WEB-3501 - Range Thresholds

henry-tp opened the pull request.

henry-tp commented on test/unit/components/useProviderConnectionPopup.test.js:
```diff
@@ -100,7 +100,7 @@ describe('useProviderConnectionPopup', function () {
 
       done();
-    }, 100);
+    }, 1000);
```
They were failing locally for me

clintonium-119 replied:
If we need to, it's fine.  Is there a number less than 1000ms that still works, like 250 or 500?  Not a big deal to add a couple of seconds to the tests, but the less we do this, the better.

If it's flaky for you with those lower values and 1000 works, I wouldn't want to to spend any extra time trying to find a sweet spot for the sake of a couple of seconds.

henry-tp commented on app/core/utils.js:
```diff
@@ -490,6 +490,7 @@ utils.roundDown = (value, precision = 0) => {
   return Math.floor(value * shift) / shift;
 };
 
+// Deprecated as of WEB-3501. See WEB-2551 for original intent and logic behind function.
```
There are no more references to this method, but I don't want to delete it. Too much work went into it, and something tells me we'll be coming back.

clintonium-119 replied:
Nah, delete it.  We can always find it in git history if we ever are asked to re-implement it.  Honestly, I prefer the new approach.  Much simpler rules, and easier to explain to others both how it works, and the reasoning behind it.

henry-tp replied:
Aye 🫡  Deleted: 469089ea0

henry-tp commented on app/pages/dashboard/PatientDrawer/CGMStatistics/index.js:
```diff
@@ -69,21 +69,23 @@ const CGMStatistics = ({ agpCGM }) => {
   const gmi        = formatDatum({ value: glucoseManagementIndicatorAGP }, 'gmi', { bgPrefs, useAGPFormat: true });
   const cov        = formatDatum({ value: coefficientOfVariation }, 'cv', { bgPrefs, useAGPFormat: true });
 
+  const roundedBgDaysWorn = bankersRound(bgDaysWorn, 1);
```
To address https://tidepoolteam.slack.com/archives/C053NJ4B2K0/p1744317040491469

clintonium-119 commented on app/pages/clinicworkspace/ClinicPatients.js:
```diff
@@ -2525,50 +2527,52 @@ export const ClinicPatients = (props) => {
       rangeName: 'extremeHigh',
     });
 
+    timeInRangeFilterOptions.reverse();
```
Tiny nit: could just define the array in the new intended order rather than keeping the original, and then immediately reversing it.

clintonium-119 commented on app/pages/dashboard/PatientDrawer/CGMStatistics/index.js:
```diff
@@ -69,21 +69,23 @@ const CGMStatistics = ({ agpCGM }) => {
   const gmi        = formatDatum({ value: glucoseManagementIndicatorAGP }, 'gmi', { bgPrefs, useAGPFormat: true });
   const cov        = formatDatum({ value: coefficientOfVariation }, 'cv', { bgPrefs, useAGPFormat: true });
 
+  const roundedBgDaysWorn = bankersRound(bgDaysWorn, 0);
```
I agree.  Whole number is preferable.  Nobody really thinks of 'days I wore something' to 2 decimal places.

clintonium-119 reviewed (approved):
Amazing work on all of this, @henry-tp !
LGTM, with a couple of minor questions/suggestions that don't need re-approval.

henry-tp commented:
@clintonium-119 

I have updated Blip and Viz, and added a new branch with modifications to Tideline. Please let me know if you're good with the changes I've made 

:)
